Possible wrong Go version of Docker image

### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/usage/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### Description of the problem

Hello! 👋 

I'm using `golangci-lint` as a container and during the process of migrating a project to the recently released version of Go 1.24, I noticed that the latest version of the `golangci-lint` container (v1.64.2 at the moment of writing this issue) is [not built using Go 1.24](https://github.com/golangci/golangci-lint/tree/08ae8055872f911072dd2c29b1454fc46a914245/build)?

It of course fails to run when using new features of Go 1.24 (e.g., `t.Context()` in tests).

### Version of golangci-lint

<details>

```console
$ docker run --rm -it golangci/golangci-lint:v1.64.2 go version
go version go1.23.6 linux/arm64
```

</details>

### Configuration

_Not relevant_

### Go environment

_Not relevant_

### Verbose output of running

<details>

```console
$ docker volume rm golangci-lint-cache
$ docker run --rm -u $(id -u):$(id -g) -t -v $(pwd):/src -w /src -v golangci-lint-cache:/tmp -e "GOLANGCI_LINT_CACHE=/tmp/golangci-lint-cache" -e "GOCACHE=/tmp/go-cache" golangci/golangci-lint:v1.64.2 golangci-lint run --fix
pkg/migration/migration_test.go:1: : # github.com/xxx/xxx/pkg/migration_test [github.com/xxx/xxx/pkg/migration.test]
pkg/migration/migration_test.go:31:25: t.Context undefined (type *testing.T has no field or method Context, but does have unexported field context)
pkg/migration/migration_test.go:34:36: t.Context undefined (type *testing.T has no field or method Context, but does have unexported field context)
pkg/migration/migration_test.go:42:33: t.Context undefined (type *testing.T has no field or method Context, but does have unexported field context)
pkg/migration/migration_test.go:52:35: t.Context undefined (type *testing.T has no field or method Context, but does have unexported field context)
pkg/migration/migration_test.go:60:32: t.Context undefined (type *testing.T has no field or method Context, but does have unexported field context)
pkg/migration/migration_test.go:72:35: t.Context undefined (type *testing.T has no field or method Context, but does have unexported field context)
pkg/migration/migration_test.go:80:34: t.Context undefined (type *testing.T has no field or method Context, but does have unexported field context)
pkg/migration/migration_test.go:92:35: t.Context undefined (type *testing.T has no field or method Context, but does have unexported field context)
pkg/migration/migration_test.go:99:34: t.Context undefined (type *testing.T has no field or method Context, but does have unexported field context)
pkg/migration/migration_test.go:109:35: t.Context undefined (type *testing.T has no field or method Context, but does have unexported field context)
pkg/migration/migration_test.go:109:35: too many errors (typecheck)
```

</details>

### A minimal reproducible example or link to a public repository

<details>
Try to lint any project that makes use of any new feature of Go 1.24 (`t.Context()` in tests for example) using the containerized version of golangci-lint.

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

--- a/build/buildx-alpine.Dockerfile
+++ b/build/buildx-alpine.Dockerfile
@@ -1,5 +1,5 @@
 # syntax=docker/dockerfile:1.4
-FROM golang:1.23-alpine
+FROM golang:1.24-alpine
 
 # related to https://github.com/golangci/golangci-lint/issues/3107
 ENV GOROOT /usr/local/go

--- a/build/buildx.Dockerfile
+++ b/build/buildx.Dockerfile
@@ -1,5 +1,5 @@
 # syntax=docker/dockerfile:1.4
-FROM golang:1.23
+FROM golang:1.24
 
 # related to https://github.com/golangci/golangci-lint/issues/3107
 ENV GOROOT /usr/local/go

